app/i18n/translator.html

```diff
@@ -126,103 +165,278 @@
         return el;
       }
 
-      function prepareTranslatedFile() {
+      function downloadTranslatedFile() {
+        saveTranslation();
         const translated = state.translated;
-        let data = Object.keys(translated).sort().map(k => `    "${k}": ${JSON.stringify(translated[k])}`).join(",\n");
+        let data = Object.keys(translated)
+          .sort()
+          .map((k) => `    "${k}": ${JSON.stringify(translated[k])}`)
+          .join(",\n");
         let json = `{\n${data}\n}\n`;
-        const file = new Blob([json], {type: "application/octet-stream"});
+        const file = new Blob([json], { type: "application/octet-stream" });
         const url = URL.createObjectURL(file);
 
         const downBtn = elID("down_translation_btn");
         downBtn.href = url;
         downBtn.download = "custom-translation.json";
       }
 
+      function isEntryDeleted(key) {
+        return (
+          state.original[state.selectedKey] === undefined &&
+          state.translated[state.selectedKey] !== undefined
```

Comment:
Haven't tested yet but just logically I think an entry is considered "deleted" if it existed in the previous English file and is missing in the new English file.

Reply:
I've simplified the condition. It checks now if the key is in the `deletedEntries` which is created in the `startTransaction` function and I did not alter the logic there.